python/dronin/telemetry.py

```diff
@@ -504,13 +539,13 @@ def service_connection(self, timeout=None):
             data = self._receive(expire)
             self.__handle_frames(data)
 
-            if len(data):
+            if self.eof:
```

Comment:
There's lots of changes to EOF behavior in here.  Previously, idle connections would return up '' instead of something EAGAIN-ish, which would be interpreted as EOF.  There's different paths for in-thread vs in-iter processing.  It could use further improvement but this seems to work well.